### PR TITLE
{Test} Fix live tests for aks, advisor, appconfig, cognitiveservices, cosmosdb, dls, hdinsight, policyinsights, iot, role, security, servicefabric

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

Fix some live test failures in aks, advisor, appconfig, cognitiveservices, cosmosdb, dls, hdinsight, policyinsights, iot, role, security, servicefabric
 
**Testing Guide**  

azdev test <module>
Result from live test pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=594754&view=artifacts&type=publishedArtifacts

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -475,14 +478,15 @@ def test_aks_create_default_service_with_virtual_node_addon(self, resource_group
...
+            'resource_type': 'Microsoft.ContainerService/ManagedClusters',
+            'subnet_id': subnet_id
         })
 
         # create cluster with virtual-node addon
         create_cmd = 'aks create --resource-group={resource_group} --name={name} --location={location} ' \
                      '--dns-name-prefix={dns_name_prefix} --node-count=1 --ssh-key-value={ssh_key_value} ' \
                      '--service-principal={service_principal} --client-secret={client_secret} --enable-addons virtual-node ' \
-                     '--aci-subnet-name foo --vnet-subnet-id bar'
+                     '--aci-subnet-name foo --vnet-subnet-id "{subnet_id}"'
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:489]: There is a validator to check that `--vnet-subnet-id` should be a resource id.

This test still fails with `Operation failed with status: 'Bad Request'. Details: ACI Connector requires azure network plugin`, leave that to AKS team to fix.

**Review by bim-msft — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_account_properties.py [hunk 2]
```diff
@@ -55,6 +55,13 @@ def test_cognitiveservices_account_public_network_access(self, resource_group):
         print(account)
         self.assertEqual(account['properties']['publicNetworkAccess'], 'Enabled')
 
+        import time
+        for i in range(10):
+            time.sleep(15)
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_account_properties.py:60]: Why 15?

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_account_properties.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/cognitiveservices/tests/latest/test_account_properties.py:60]: In another similar cognitiveservices test, a comment says 
> when generating recording, use a large value such as 15

We can put the wait and retry logic in testsdk later so we don't need to write the loop ourselves.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 3]
```diff
@@ -334,9 +334,11 @@ def test_graph_group_scenario(self):
             return  # this test delete users which are beyond a SP's capacity, so quit...
 
         domain = username.split('@', 1)[1]
+        if domain == 'microsoft.com' and self.is_live:
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:337]: Actually, our live run CI uses an `@AzureSDKTeam.onmicrosoft.com` account and won't trigger this issue.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py [hunk 4]
```diff
@@ -334,9 +334,11 @@ def test_graph_group_scenario(self):
...
+            domain = self.cmd('rest --method get --url https://graph.windows.net/myorganization/domains?api-version=1.6 --query "value[?isVerified==`true`].name|[0]"').get_output_in_json()
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:338]: Another approach would be to call command 

```
> az ad signed-in-user show
{
  ...
  "userPrincipalName": "rolelivetest@azuresdkteam.onmicrosoft.com",
```

I tried this before but reverted the change given I always use a `@AzureSDKTeam.onmicrosoft.com` account.

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:338]: This domain is also a verified domain but is not a valid email address:

```
    {
      "authenticationType": "Managed",
      "availabilityStatus": null,
      "forceDeleteState": null,
      "isAdminManaged": true,
      "isDefault": false,
      "isDefaultForCloudRedirections": false,
      "isInitial": false,
      "isRoot": true,
>>    "isVerified": true,
>>    "name": "azdevextest.com",
      "passwordNotificationWindowInDays": null,
      "passwordValidityPeriodInDays": null,
      "state": null,
      "supportedServices": []
    }
```

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py (hunk 3 above)
Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py:337]: You are right. I found this issue while fixing [another error](https://clitestresultstac.blob.core.windows.net/20201027-111238-748168/role.report.parallel.html) locally.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py (hunk 4 above)
Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py]: Reverted as other role tests with `@microsoft.com` username may have the same issue and tests should be done with `@AzureSDKTeam.onmicrosoft.com` account for now.

**Review by bim-msft — APPROVED** [review]

**Review by Juliehzl — APPROVED** [review]
Approve for code owner part

**Pull request merged**